data/transactions/logic/mocktracer/tracer.go

```diff
@@ -112,8 +115,8 @@ func AfterTxn(txnType protocol.TxType, ad transactions.ApplyData, hasError bool)
 }
 
 // AfterProgram creates a new Event with the type AfterProgramEvent
-func AfterProgram(mode logic.RunMode, hasError bool) Event {
-	return Event{Type: AfterProgramEvent, LogicEvalMode: mode, HasError: hasError}
+func AfterProgram(mode logic.RunMode, pass bool, hasError bool) Event {
```

Comment:
suggestion: two boolean flags in a row do not look great, maybe invent a small `programStatus` type with values "OK" (pass=T, err=F), "REJ" (pass=F, err=F), "ERR" (pass=F, err=T) - assuming pass=T, err=T is impossible combo.

Reply:
I like this, it's easier to understand. Made change in 8fd295f56